app/src/main/java/io/homeassistant/companion/android/onboarding/integration/MobileAppIntegrationView.kt

```diff
@@ -97,6 +108,72 @@ fun MobileAppIntegrationView(
                     fontWeight = FontWeight.Light
                 )
             }
+            if (onboardingViewModel.deviceIsWatch && onboardingViewModel.mayRequireTlsClientCertificate) {
+                Spacer(modifier = Modifier.height(16.dp))
+
+                Text(
+                    text = stringResource(id = commonR.string.tls_cert_onboarding_title),
+                    style = MaterialTheme.typography.h6
+                )
+                Text(
+                    text = stringResource(id = commonR.string.tls_cert_onboarding_description),
+                    fontWeight = FontWeight.Light
+                )
+                Row(
+                    modifier = Modifier.fillMaxWidth(),
+                    horizontalArrangement = Arrangement.SpaceBetween
+                ) {
+                    Button(onClick = onSelectTLSCertificateClicked) {
+                        Text(text = stringResource(id = commonR.string.select_file))
+                    }
+                    Text(
+                        text = onboardingViewModel.tlsClientCertificateFilename,
+                        modifier = Modifier
+                            .align(Alignment.CenterVertically)
+                    )
+                }
+                if (onboardingViewModel.tlsClientCertificateUri != null) {
+                    Row {
+                        TextField(
+                            value = onboardingViewModel.tlsClientCertificatePassword,
+                            onValueChange = {
+                                onboardingViewModel.tlsClientCertificatePassword = it
+                                onCheckPassword(onboardingViewModel.tlsClientCertificatePassword)
```

Comment:
This technically works but typing is sluggish because it is checking the password for every letter you enter. You should use a local variable for the text value and have it check the password in the background instead.

Reply:
@jpelgrom I moved the password check to the ViewModel coroutine scope. Is this correct?

Reply:
Works but I'd use `lifecycleScope` from the fragment to keep it simpler.

Reply:
I changed it to lifecycleScope.